src/Deferred.php

```diff
@@ -13,16 +13,34 @@
  */
 final class Deferred implements Promise
 {
+    /**
+     * @var ResponseInterface|null
+     */
     private $value;
 
+    /**
+     * @var Exception|null
```

Comment:
should this be Psr\Http\Client\ClientExceptionInterface ? (i see that for now the type declaration in the method is the httplug exception interface, but psr exception would be future proof...

Reply:
You are correct. Thanks